metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestRolesStep.java

```diff
@@ -0,0 +1,162 @@
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.entity.EnvelopedAspect;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.metadata.boot.BootstrapStep;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.models.AspectSpec;
+import com.linkedin.metadata.models.registry.EntityRegistry;
+import com.linkedin.metadata.query.ListUrnsResult;
+import com.linkedin.metadata.search.EntitySearchService;
+import com.linkedin.metadata.search.transformer.SearchDocumentTransformer;
+import com.linkedin.metadata.search.utils.SearchUtils;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.GenericAspect;
+import com.linkedin.mxe.MetadataChangeProposal;
+import com.linkedin.policy.DataHubRoleInfo;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.ClassPathResource;
+
+import static com.linkedin.metadata.Constants.*;
+
+
+@Slf4j
+@RequiredArgsConstructor
+public class IngestRolesStep implements BootstrapStep {
+
+  private final EntityRegistry _entityRegistry;
+  private final EntityService _entityService;
+  private final EntitySearchService _entitySearchService;
+  private final SearchDocumentTransformer _searchDocumentTransformer;
+
+  @Override
+  public String name() {
+    return this.getClass().getSimpleName();
+  }
+
+  @Override
+  public void execute() throws Exception {
+    final ObjectMapper mapper = new ObjectMapper();
+
+    // 0. Execute preflight check to see whether we need to ingest Roles
+    log.info("Ingesting default Roles...");
+
+    // 1. Read from the file into JSON.
+    final JsonNode rolesObj = mapper.readTree(new ClassPathResource("./boot/roles.json").getFile());
+
+    if (!rolesObj.isArray()) {
+      throw new RuntimeException(
+          String.format("Found malformed roles file, expected an Array but found %s", rolesObj.getNodeType()));
+    }
+
+    for (final JsonNode roleObj : rolesObj) {
+      final Urn urn = Urn.createFromString(roleObj.get("urn").asText());
+
+      // If the info is not there, it means that the role was there before, but must now be removed
+      if (!roleObj.has("info")) {
+        _entityService.deleteUrn(urn);
+        continue;
+      }
+
+      final DataHubRoleInfo info = RecordUtils.toRecordTemplate(DataHubRoleInfo.class, roleObj.get("info").toString());
+      ingestRole(urn, info);
+    }
+    // If search index for roles is empty, update the role index with the ingested roles from previous step.
+    // Directly update the ES index, does not produce MCLs
+    if (_entitySearchService.docCount(DATAHUB_ROLE_ENTITY_NAME) == 0) {
```

Comment:
We don't need this block - please remove. This was done for historical reasons.

Reply:
Removed